beanie/odm/models.py

```diff
@@ -11,7 +11,7 @@ class InspectionError(BaseModel):
     Inspection error details
     """
 
-    document_id: PydanticObjectId
+    document_id: Union[PydanticObjectId, UUID4]
```

Comment:
since you can technicaly overide id (needs to be tested) with anything that is serializable by beanie... this is  is probably not enough (we should probably create Union of  Anything serializable probably something here https://github.com/BeanieODM/beanie/blob/54e0f2d5b2f3e9c83f37393fdd8bde4bf38fde81/beanie/odm/utils/encoder.py#L30 ) (somewhere close to Here
we have str as id and this would not work

I also do not see any tests  and it appears there is techical Debt in InspectionErrorCoverage


it is however coverable...  you insert with another library somethig that would triger InspectionError

second atitional test is to test if such type can be used as id (so use union of serializable to do that)